packages/peregrine/src/hooks/__tests__/useDOMQuery.spec.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import {
+    TestValidComponent,
+    TestInvalidComponent
+} from '../__mocks__/useDOMQuery';
+
+describe('Unit testing useDOMQuery hook.', () => {
+    beforeAll(() => {
+        const div = document.createElement('div');
+        window.domNode = div;
+        document.body.appendChild(div);
+    });
+
+    it('setInnerText, setInnerHTML and setAttribute manipulate DOM when a valid query selector is provided', () => {
+        const newText = 'Changed after mount.';
+        const newHTML = '<i>Changed after mount.</i>';
+        const newName = 'newName';
+        const wrapper = mount(
+            <TestValidComponent
+                newText={newText}
+                newHTML={newHTML}
+                newName={newName}
+            />,
+            { attachTo: window.domNode }
+        );
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve();
+            }, 0);
+        }).then(() => {
```

Comment:
There has to be a better way to do this 😆

Reply:
I'm assuming this will go away after removing `Enzyme` anyway

Reply:
Unfortunately no 😭

Reply:
Cleaned it up. If in the future we can find a cleaner solution to this, it will be easy to delete unnecessary code.